role_manager_b_test.go

```diff
@@ -0,0 +1,95 @@
+package casbin
+
+import (
+	"fmt"
+	"testing"
+
+	defaultrolemanager "github.com/casbin/casbin/v2/rbac/default-role-manager"
+)
+
+func BenchmarkRoleManagerSmall(b *testing.B) {
+	e, _ := NewEnforcer("examples/rbac_model.conf", false)
+	// Do not rebuild the role inheritance relations for every AddGroupingPolicy() call.
+	e.EnableAutoBuildRoleLinks(false)
+
+	// 100 roles, 10 resources.
+	pPolicies := make([][]string, 0)
+	for i := 0; i < 100; i++ {
+		pPolicies = append(pPolicies, []string{fmt.Sprintf("group%d", i), fmt.Sprintf("data%d", i/10), "read"})
+	}
+	e.AddPolicies(pPolicies)
+
+	// 1000 users.
+	gPolicies := make([][]string, 0)
+	for i := 0; i < 1000; i++ {
+		gPolicies = append(gPolicies, []string{fmt.Sprintf("user%d", i), fmt.Sprintf("group%d", i/10)})
+	}
+
+	e.BuildRoleLinks()
+
+	rm := defaultrolemanager.NewRoleManager(1)
+
+	for i := 0; i < b.N; i++ {
+		for j := 0; j < 100; j++ {
```

Comment:
I think it's better to use random number, like `rm.HasLink("user501", fmt.Sprintf("group%d", rand.Intn(100)))`. 

What do you think?

Reply:
I reference https://github.com/casbin/casbin-rs/blob/b3bd688fd7fbb3b0c497b0ad255e129ed549b945/benches/benchmark.rs#L144, and he seems to use a loop. And he wants to compare the casbin's bench with casbin-rs's bench

Reply:
loop there is to simulate enforce.

Because what we need to know is: *how much time has been taken up by role manager during an enforce*.